forem/forem
# Raise a clearer error message when the user tries to comment on a draft post

## What type of PR is this? (check all applicable)


## Description
The user experience when submitting an invalid comment on a draft PR is really confusing. They get the error message "Your comment could not be posted due to an error: Commentable is not valid." This PR updates the error message to "Your comment could not be posted due to an error: Commentable is not a published article"

## Related Tickets & Documents



- Related Issue #18386
- Closes #18386

## QA Instructions, Screenshots, Recordings

- Go to `/new`
- Fill in the title
- Click Save Draft
- Fill in the comment text area for that unpublished article
- Click submit on the comment
- [ ] Clearer Error message should be displayed

![image](https://user-images.githubusercontent.com/8124558/196053764-594f7054-5da0-4245-849f-95034f713151.png)


### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![gif selected by bribery from jesus_h](https://media.giphy.com/media/3EiNpweH34XGoQcq9Q/giphy.gif)
